src/tdvp.jl

```diff
@@ -71,11 +71,11 @@ Optional keyword arguments:
 * `write_when_maxdim_exceeds::Int` - when the allowed maxdim exceeds this value, begin saving tensors to disk to free memory in large calculations
 """
 function tdvp(H, t::Number, psi0::MPS; kwargs...)
-  return sweep_update(tdvp_solver(; kwargs...), H, t, psi0; kwargs...)
+  return alternating_update(tdvp_solver(; kwargs...), H, t, psi0; kwargs...)
 end
 
 function tdvp(solver, H, t::Number, psi0::MPS; kwargs...)
```

Comment:
Are we keeping this function around? Basically this is saying that `tdvp` is an alias for `alternating_update`.

The reason I can think of for not allowing this is to keep a more strict definition of `tdvp` as a time evolution algorithm.

The reason I can think of for keeping it is that we do want a generic version of `tdvp` that allows users to input different solvers (like an ODE solver from DifferentialEquations.jl, different Krylov-based exponentiation backends, etc.), but maybe that should have a different interface. That is already done to some extent by choosing `applyexp` vs. `KrylovKit.exponentiate` through keyword arguments, so I think the question here is what we want our interface to be for an extensible backend system for TDVP solvers.

Reply:
Good catch. After thinking about your points I agree let's remove this for now. I think your point is the right one that we'll offer a more "controlled" interface for custom solvers, say to make sure they match the algorithm being requested (i.e. that they are all TDVP solvers). We can also certainly revisit having exposing this interface to users again some other way in the future, so no harm in removing it now to simplify things.

Reply:
Resolved in latest commit